Levi's first draft

levikl opened the pull request:
@jaybobo

Commit:
Levi's first draft

mikegee commented on challenge.rb:
```diff
@@ -1,26 +1,26 @@
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all { |animal| animal == "dog" }
```
`select` is an alias of `find_all`, and preferred by most Rubyists.

mikegee commented on challenge.rb:
```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |string| string.capitalize }
```
There's a shorthand that's equivalent but a bit "magical": `input.map(&:capitalize)`

The shorthand is typically preferred.

Check out http://www.brianstorti.com/understanding-ruby-idiom-map-with-symbol/ or Google "ruby symbol to_proc" for others.

mikegee commented on challenge.rb:
```diff
@@ -1,26 +1,26 @@
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
```
`first(2)` would work here too.

Commit:
Apply all recommended changes

levikl commented:
Also changed ```fetch_CD_animals``` to use ```grep``` since I changed ```fetch_the_dog``` to use ```select```

(Per README: "Don't duplicate methods for each function. If you use ```select``` once, don't use it again for the challenge.")

mikegee commented:
> Don't duplicate methods for each function. If you use select once, don't use it again for the challenge."

TIL
